csa.rs

```diff
@@ -0,0 +1,134 @@
+use std::io;
+
+const MAX_STATIONS: usize = 100000;
+
+#[derive(Debug)]
+struct Connection {
+    departure_station: usize,
+    arrival_station: usize,
```

Comment:
Actually, it's something I don't understand; it seems that I can only index an array using `usize`. When I tried to use `u32` everywhere, I had an error about arrays not being indexable using `u32`.